app/App.hs

```diff
@@ -77,10 +80,29 @@ runAppIO args@RunAppIOArgs {..} =
     ExitJuvixError e -> do
       printErr e
       embed exitFailure
-    ExitMsg exitCode t -> embed (putStrLn t >> hFlush stdout >> exitWith exitCode)
+    ExitMsg exitCode t -> exitMsg' exitCode t
     SayRaw b -> embed (ByteString.putStr b)
   where
+    exitMsg' :: ExitCode -> Text -> Sem r x
+    exitMsg' exitCode t = embed (putStrLn t >> hFlush stdout >> exitWith exitCode)
+    getMainFile' :: Maybe (AppPath File) -> Sem r (Path Abs File)
+    getMainFile' m = case m of
+      Just p -> embed (prepathToAbsFile invDir (p ^. pathPath))
+      Nothing -> case pkg ^. packageMain of
+        -- TODO do something special if it is the global package?
+        Just p -> embed (prepathToAbsFile invDir p)
+        Nothing -> missingMainErr
+    missingMainErr :: Sem r x
+    missingMainErr =
+      exitMsg'
+        (ExitFailure 1)
+        ( "A path to the main file must be given in the CLI or specified in the `main` field of the "
```

Comment:
```suggestion
        ( "File path to the main file is not provided. Ensure to specify it in the `main` field of the "
```

Reply:
I can't see that this improves the error message. Moreover, it is incorrect, because it does not mention the CLI